x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/add_source/github_via_app.tsx

```diff
@@ -0,0 +1,127 @@
+    useValues(GithubViaAppLogic);
+  const {
+    setGithubAppId,
+    setGithubEnterpriseServerUrl,
+    setStagedPrivateKey,
+    createContentSource,
+    setSourceIndexPermissionsValue,
+  } = useActions(GithubViaAppLogic);
+
+  const { hasPlatinumLicense } = useValues(LicensingLogic);
+  const name = isGithubEnterpriseServer ? SOURCE_NAMES.GITHUB_ENTERPRISE : SOURCE_NAMES.GITHUB;
+  const data = staticSourceData.find((source) => source.name === name);
+  const Layout = isOrganization ? WorkplaceSearchPageTemplate : PersonalDashboardLayout;
+
+  const handlePrivateKeyUpload = async (files: FileList | null) => {
+    if (!files || files.length < 1) {
+      return null;
+    }
+    const file = files[0];
+    const text = await readUploadedFileAsText(file);
+
+    setStagedPrivateKey(text);
+  };
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+    createContentSource();
+  };
+
+  return (
+    <Layout pageChrome={[NAV.SOURCES, NAV.ADD_SOURCE, name || '...']} isLoading={false}>
+      <form onSubmit={handleSubmit}>
+        <EuiPanel paddingSize="none" hasShadow={false} color="subdued">
+          <EuiPanel hasShadow={false} paddingSize="l" color="subdued">
+            <AddSourceHeader
+              name={name}
+              serviceType="github"
+              categories={['Software', 'Version Control', 'Code Repository']} // TODO: get from API
+            />
+          </EuiPanel>
+          <EuiHorizontalRule margin="xs" />
+          <EuiPanel hasShadow={false} paddingSize="l" color="subdued">
+            <SourceFeatures features={data!.features} name={name} objTypes={data!.objTypes} />
+          </EuiPanel>
+        </EuiPanel>
+
+        <EuiSpacer />
+
+        {!hasPlatinumLicense && <DocumentPermissionsCallout />}
+        {hasPlatinumLicense && isOrganization && (
+          <DocumentPermissionsField
+            needsPermissions
+            indexPermissionsValue={indexPermissionsValue}
+            setValue={setSourceIndexPermissionsValue}
+          />
+        )}
+
+        <EuiFormRow label="GitHub App ID">
+          <EuiFieldText value={githubAppId} onChange={(e) => setGithubAppId(e.target.value)} />
+        </EuiFormRow>
+        {isGithubEnterpriseServer && (
+          <EuiFormRow label="Base URL">
+            <EuiFieldText
+              value={githubEnterpriseServerUrl}
+              onChange={(e) => setGithubEnterpriseServerUrl(e.target.value)}
+            />
+          </EuiFormRow>
+        )}
+        <EuiFormRow label="Private key" helpText="Upload private key (.pem) to authenticate GitHub">
+          <EuiFilePicker onChange={handlePrivateKeyUpload} accept=".pem" />
+        </EuiFormRow>
+        <EuiButton fill type="submit" isLoading={isSubmitButtonLoading}>
+          {isSubmitButtonLoading
+            ? 'Connecting…'
+            : `Connect ${isGithubEnterpriseServer ? 'GitHub Enterprise Server' : 'GitHub'}`}
```

Comment:
Can we use the constants in this file?
```suggestion
            ? 'Connecting…'
            : `Connect ${isGithubEnterpriseServer ? SOURCE_NAMES.GITHUB_ENTERPRISE : SOURCE_NAMES.GITHUB}`}
```